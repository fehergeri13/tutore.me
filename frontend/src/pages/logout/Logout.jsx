import React from 'react';
import {inject, observer} from 'mobx-react';
import autobind from 'autobind-decorator';
import {observable, toJS} from 'mobx';

import "./logout.less"


@inject('model')
@observer
export default class Logout extends React.Component {

    componentDidMount() {
        this.props.model.auth.doLogout();
    }

    render() {
        return <div className="logout-page">
            <h2>Logging out...</h2>
        </div>;
    }
}
