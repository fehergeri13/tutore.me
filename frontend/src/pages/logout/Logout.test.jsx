import React from 'react';
import renderer from 'react-test-renderer';
import Logout from "./Logout";
import {Provider} from 'mobx-react';

describe('Logout', () => {
    it('renders correctly', () => {
        const mockModel = {
            auth: {
                doLogout() {},
            }
        };
        const tree = renderer.create(
            <Provider model={mockModel}><Logout /></Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
