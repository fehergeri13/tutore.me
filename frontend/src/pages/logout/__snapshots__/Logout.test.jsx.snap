// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout renders correctly 1`] = `
<div
  className="logout-page"
>
  <h2>
    Logging out...
  </h2>
</div>
`;
